app/src/main/java/org/thoughtcrime/securesms/conversation/v2/input_bar/InputBarRecordingView.kt

```diff
@@ -132,7 +132,24 @@ class InputBarRecordingView : RelativeLayout {
     private fun updateTimer() {
         val duration = (Date().time - startTimestamp) / 1000L
         binding.recordingViewDurationTextView.text = DateUtils.formatElapsedTime(duration)
-        snHandler.postDelayed({ updateTimer() }, 500)
+
+        if (isAttachedToWindow) {
+            // Make sure there's only one runnable in the handler at a time.
+            removeCallbacks(updateTimerRunnable)
+
+            // Should only update the timer if the view is still attached to the window.
+            // Otherwise, the timer will keep running even after the view is detached.
+            postDelayed(updateTimerRunnable, 500)
+        }
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        if (isVisible) {
```

Comment:
My intention is to fix the memory leak and potential bug that animation won't run on reattach. What you are saying is a performance future proof (i.e. prevent excessive timer run) which I think doesn't need to happen in this PR.

To be able to control the timer correctly I think you'd need to do the timer thing in `dispatchOnDraw`, which I think it's too big a rabbit hole to get into.

Reply:
Or rewrite in compose haha..

Reply:
I'm saying that now this animation will start when the activity starts, because when this view is inflated and attached, it is `visible`, even though it's parent layout is not, so the manual call to `updateTimer` is basically a no-op, because `updateTimer` is already running.

It'd be nice for this animation to not run while the Activity is in the background.